wallet migrated to version 1.0.0-beta.2

tima-t opened the pull request.

monokh reviewed (commented):
I'm a bit confused on the fee situation.

When I send a NEAR tx,  does the fee come out of my wallet? But the fee is displayed as 0
Everything else is working fine

tima-t commented on src/utils/asset.js:
```diff
@@ -90,3 +101,11 @@ export const getAssetIcon = (asset) => {
+export const getExplorerTransactionHash = (asset, hash) => {
+  switch (asset) {
+    case 'NEAR':
+      return hash.split('_')[0]
```
Yeah It is explorer specific. The problem is that in the Near package the transaction hash is stored in the form TXHASH_ address as this is the right format.   Not sure if it could be included in CAL

tima-t commented on src/store/factory/client.js:
```diff
@@ -113,6 +125,22 @@ function createEthClient (asset, network, mnemonic, walletType) {
+  nearClient.addProvider(new NearSwapProvider())
+  nearClient.addProvider(new NearSwapFindProvider(nearNetwork?.helperUrl))
+  nearClient.assertValidTransaction = function (transaction) { }
+  nearClient.chain.getTransactionByHash = async (txHash) => {
```
Yes, this should be moved in CAL, but maybe it is already fixed, I will test and remove this code

kraikov commented:
> I'm a bit confused on the fee situation.
> 
> When I send a NEAR tx, does the fee come out of my wallet? But the fee is displayed as 0
> Everything else is working fine

Unfortunately it's not a trivial task to calculate the fee. For example if you try to use the official near wallet, you will see they do not display the fees, but instead reserve 0.35 Near for fee.

monokh commented:
I'm noticing a bug. If i updated my wallet to this version, then go to Manage Assets and enable NEAR. It does not appear and no errors in console

monokh commented:
The NEAR logo is not matching other assets in style

tima-t commented on package.json:
```diff
@@ -18,29 +18,34 @@
+    "@liquality/ethereum-erc20-provider": "^1.0.0-beta.2",
+    "@liquality/ethereum-erc20-scraper-swap-find-provider": "^1.0.0-beta.2",
+    "@liquality/ethereum-erc20-swap-provider": "^1.0.0-beta.2",
+    "@liquality/ethereum-gas-station-fee-provider": "0.10.0",
```
Package changed and version updated

tima-t commented on src/store/factory/client.js:
```diff
@@ -113,6 +125,22 @@ function createEthClient (asset, network, mnemonic, walletType) {
+  nearClient.addProvider(new NearJsWalletProvider(nearNetwork, mnemonic))
+  nearClient.addProvider(new NearSwapProvider())
+  nearClient.addProvider(new NearSwapFindProvider(nearNetwork?.helperUrl))
+  // nearClient.assertValidTransaction = function (transaction) { }
```
Removed

tima-t commented:
> The NEAR logo is not matching other assets in style

Can we ask your designers to prepare such logo in svg format as I cannot find such on the web

tima-t commented:
> I'm noticing a bug. If i updated my wallet to this version, then go to Manage Assets and enable NEAR. It does not appear and no errors in console

Maybe this was a problem related to cryptoassets package. 
I've updated the package and check everything locally and it works fine now. 
Cannot reproduce the bug anymore
